src/scenario.rs

```diff
@@ -0,0 +1,462 @@
+	DefaultResubmissionQueue<DefaultTxTask<EthTransaction>>,
+>;
+pub(crate) struct EthScenarioExecutor {
+	stop_sender: Sender<()>,
+	runner: EthScenarioRunner,
+	resubmission_queue: ResubmissionQueueTask,
+}
+
+pub(crate) type SubstrateScenarioRunner = Runner<
+	DefaultTxTask<SubstrateTransaction>,
+	SubstrateTransactionsSink,
+	DefaultResubmissionQueue<DefaultTxTask<SubstrateTransaction>>,
+>;
+pub(crate) struct SubstrateScenarioExecutor {
+	stop_sender: Sender<()>,
+	runner: SubstrateScenarioRunner,
+	resubmission_queue: ResubmissionQueueTask,
+}
+
+impl SubstrateScenarioExecutor {
+	pub(crate) fn new(
+		stop_sender: Sender<()>,
+		runner: SubstrateScenarioRunner,
+		resubmission_queue: ResubmissionQueueTask,
+	) -> Self {
+		SubstrateScenarioExecutor { stop_sender, runner, resubmission_queue }
+	}
+}
+
+impl EthScenarioExecutor {
+	pub fn new(
+		stop_sender: Sender<()>,
+		runner: EthScenarioRunner,
+		resubmission_queue: ResubmissionQueueTask,
+	) -> Self {
+		EthScenarioExecutor { stop_sender, runner, resubmission_queue }
+	}
+}
+
+/// Multi-chain scenario executor.
+pub enum ScenarioExecutor {
+	Eth(EthScenarioExecutor),
+	Substrate(SubstrateScenarioExecutor),
+}
+
+impl ScenarioExecutor {
+	pub async fn execute_txs<T: TxTask + 'static>(
+		self,
+	) -> HashMap<H256, Arc<DefaultExecutionLog<H256>>> {
+		match self {
+			ScenarioExecutor::Eth(mut inner) => {
+				let (_, logs) =
+					futures::future::join(inner.resubmission_queue, inner.runner.run()).await;
+				logs
+			},
+			ScenarioExecutor::Substrate(mut inner) => {
+				let (_, logs) =
+					futures::future::join(inner.resubmission_queue, inner.runner.run()).await;
+				logs
+			},
+		}
+	}
+
+	/// Installs a ctrl_c handler which sends a stop notification on the executor
+	/// stop sender channel, to notify the stop of the scenario for displaying partial stats about
+	/// the transactions execution.
+	///
+	/// Can be called only once for the lifetime of a the program execution.
+	fn install_ctrlc_stop_hook(&self) {
+		let stop_sender = match &self {
+			ScenarioExecutor::Eth(inner) => inner.stop_sender.clone(),
+			ScenarioExecutor::Substrate(inner) => inner.stop_sender.clone(),
+		};
+		ctrlc::set_handler(move || {
+			block_on(stop_sender.send(())).expect("Could not send signal on channel.")
+		})
+		.expect("Error setting Ctrl-C handler");
+	}
+}
+/// Building logic for the execution of a scenario.
+#[derive(Default)]
+pub struct ScenarioBuilder {
+	scenario_type: Option<ScenarioType>,
+	tx_recipe: Option<TransactionRecipe>,
+	does_block_monitoring: bool,
+	watched_txs: bool,
+	send_threshold: Option<usize>,
+	rpc_uri: Option<String>,
+	chain_type: Option<ChainType>,
+	installs_ctrl_c_stop_hook: bool,
+}
+
+impl ScenarioBuilder {
+	pub fn with_scenario_type(mut self, scenario_type: ScenarioType) -> Self {
+		self.scenario_type = Some(scenario_type);
+		self
+	}
+
+	pub fn with_transfer_tx_recipe(mut self) -> Self {
+		self.tx_recipe = Some(TransactionRecipe::transfer());
```

Comment:
This should be default.

Reply:
Yes, added it by default when constructing the scenario, and can be changed if `remark` is set. Change here: [2319bdf](https://github.com/michalkucharczyk/tx-test-tool/pull/22/commits/2319bdf4509881f8c4c8bf186444a2b605fdb8d5)